source/tutorials/cross-compilation.rst

```diff
@@ -0,0 +1,262 @@
+.. _ref-cross-compilation:
+
+Cross-compilation
+=================
+
+Cross-compilation is the act of **compiling code** on the **build platform**
+to the **host platform**, where the compiled **executable runs**. [#]_
+
+It's needed when the host platform has limited resources (such as CPU)
+or when it's not easily accessible for development.
+
+The Nix community has world-class support for cross-compilation,
+after years of hard work from our community.
+
+.. [#] Terminology for cross-compilation platforms differs between build systems.
+       We have chosen to follow 
+       `autoconf terminology <https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Hosts-and-Cross_002dCompilation.html>`_.
+
+.. note:: macOS/Darwin is a special case, as not the whole OS is open-source. 
+          It's only possible to cross-compile between ``aarch64-darwin`` and ``x86_64-darwin``.
+
+
+What's a target platform?
+-------------------------
+
+There's actually a third platform named the target platform.
+
+It matters in cases where you'd like to distribute a compiler binary, 
+as you'd then like to build a compiler on the build platform, compile code on the
+host plaform and run the final executable on the target platform.
+
+
+Since that's rarely needed, we'll treat the target platform the same as the host.
+
+
+Determining the host platform config
+------------------------------------
+
+The build platform is determined automatically by Nix
+as it can just guess it during the configure phase.
+
+The host platform is best determined by running on the host platform:
+
+.. code:: shell-session 
+
+  $ bash $(nix-build '<nixpkgs>' -A gnu-config)/config.guess
+  aarch64-unknown-linux-gnu
+
+In case that's not possible (when the host platform is not easily accessible
+for development), the platform config has to be constructed manually via the following template:
+
+.. code::
+
+  <cpu>-<vendor>-<os>-<abi>
+
+Note that ``<vendor>`` is often ``unknown`` and ``<abi>`` is optional. 
+There's also no unique identifier for a platform, for example ``unknown`` and 
+``pc`` are interchangeable (hence it's called config.guess).
+
+If you can't install Nix, find a way to run ``config.guess`` (usually comes with
+ the autoconf package) from the OS you're able to run on the host platform.
+
+Some other common examples of platform configs:
+
+- aarch64-apple-darwin14
+- aarch64-pc-linux-gnu
+- x86_64-w64-mingw32
+- aarch64-apple-ios
+
+
+Choosing the host platform with Nix
+-----------------------------------
+
+Nixpkgs comes with a set of predefined host platforms applied to all packages.
+
+It's possible to list predefined attribute sets via shell completion:
```

Comment:
Shell completion doesn't always work fully, so it might be nice to give the command to list these as well. I'm often scratching my head when thinking about how to search for things in nixpkgs so it's good to be reminded often I feel

Reply:
Good point! I should maybe show how to use a reply here.

Reply:
`lib/systems/examples.nix` is not a scary file I think it is usable as a list of platforms.

Reply:
I'd like to avoid pointing the user to a file (through github) as a reference to what attributes `pkgsCross` has as it's hard to explain the connection between the two :(

Reply:
I mean the alternative is telling them to run

```console
$ nix-instantiate --eval --strict -E 'builtins.attrNames (import <nixpkgs/lib>).systems.examples'
```

pretty much…

Reply:
Went with `nix repl` as that's a good thing to teach anyway (although it adds more things to introduce here, it's for the better).